lib/themes/dosomething/paraneue_dosomething/js/utilities/LazyLoader.js

```diff
@@ -0,0 +1,50 @@
+import Layzr from 'layzr.js';
+
+// Application lazy loader instance.
+let loader = null;
+
+/**
+ * Helper function to check for new [data-src] elements that were added
+ * after DOMContentLoaded and lazy-load them.
+ */
+export function update() {
+  if (loader === null) {
+    throw new Error('Lazy loader has not been initialized.');
+  }
+
+  loader.update().check();
+}
+
+/**
+ * Initialize a new lazy loader.
+ *
+ * @returns Layzr
+ */
+export function init() {
+  if (loader !== null) {
+    return loader;
+  }
+
+  loader = new Layzr({
+    normal: 'data-src',
+    retina: 'data-src-retina',
+  });
+
+  loader.on('src:before', image => {
+    // On load, set the images to be transparent so we can fade them in.
+    image.style.transition = 'opacity 0.5s ease-in';
+    image.style.opacity = 0;
```

Comment:
I originally did that with `.lazy-loaded` and `.is-loaded` classes, but then it seemed ridiculous to have to hunt down everywhere this applies to add the class. I don't have strong feelings, but it does feel kinda nice and tidy to have everything in this one little file.